src/Microsoft.AspNetCore.Http.Connections.Client/HttpConnection.cs

```diff
@@ -115,17 +115,21 @@ internal HttpConnection(HttpConnectionOptions httpConnectionOptions, ILoggerFact
             _transportFactory = transportFactory;
         }
 
-        public async Task StartAsync()
+        public Task StartAsync() => StartAsync(TransferFormat.Binary, CancellationToken.None);
+
+        public Task StartAsync(CancellationToken cancellationToken) => StartAsync(TransferFormat.Binary, cancellationToken);
+
+        public async Task StartAsync(TransferFormat transferFormat)
```

Comment:
Can we get rid of this overload by by changing the next one to use `CancellationToken cancellationToken = default`

Reply:
Can I get clarity on our policy with overloads vs default arguments @davidfowl ? Previously I'd heard that default arguments are "bad" because they are a language-specific feature and it depends on the caller's compiler filling in the default. I'm happy to use them if we don't care about that. It makes a ton of things easier.

Reply:
I see we've got https://github.com/aspnet/Home/wiki/Engineering-guidelines#async-method-patterns